### PR TITLE
authtool: Enhance argument combinations validation

### PR DESCRIPTION
**Pull request by badone**
Under certain circumstances ceph-authtool can create invalid key entries in
the specified keyring and behave less than intuitively. What's worse these
keys can currently cause ceph daemons to crash. This patch attempts to tighten
the behaviour and eliminate the possibility an invalid key can be created.

http://tracker.ceph.com/issues/2904

Fixes: #2904
Signed-off-by: Brad Hubbard <bhubbard@redhat.com>

### CODE REVIEW

**Comment by badone** [comment]
make[5]: *** Deleting file `test/cephtool-test-mds.sh.log'

Known issue.

File: src/tools/ceph_authtool.cc [hunk 1]
```diff
@@ -86,6 +86,10 @@ int main(int argc, const char **argv)
     } else if (ceph_argparse_flag(args, i, "--gen-print-key", (char*)NULL)) {
       gen_print_key = true;
     } else if (ceph_argparse_witharg(args, i, &val, "-a", "--add-key", (char*)NULL)) {
+      if (val.empty()) {
+        cerr << "Option --add-key requires an argument" << std::endl;
```
Comment (tchaikov) [inline src/tools/ceph_authtool.cc]: maybe "an non-empty argument".

Comment (badone) [inline src/tools/ceph_authtool.cc]: Well, this matches an existing error message from ceph_argparse for consistency.

Comment (tchaikov) [inline src/tools/ceph_authtool.cc]: got it!

File: src/test/cli/ceph-authtool/add-key.t [hunk 2]
```diff
@@ -12,3 +12,9 @@
   $ cat kring
   [client.admin]
   \tkey = AQAK7yxNeF+nHBAA0SgSdbs8IkJrxroDeJ6SwQ== (esc)
+
+Test --add-key with empty argument
+
+  $ /home/brad/working/src/ceph/src/ceph-authtool kring -C --name=mon.* --add-key= --cap mon 'allow *'
```
Comment (tchaikov) [inline src/test/cli/ceph-authtool/add-key.t]: might want to use a path also applies under other circumstances.

**Comment by badone** [comment]
OOPs! Sure, let me fix that...

**Comment by tchaikov** [comment]
lgtm, and tested locally, will merge once jenkins finishes.

**Pull request merged**